### ISSUE TITLE
Extend Graph Grammar for Zelda

### ISSUE THREAD

**Issue by schrum2** [schrum2/MM-NEAT#445]
After completing #444 you should attempt extending the graph grammar to create a more interesting variety/diversity of levels. You can play around with this issue a bit ... I may make suggestions for specific rules as you work on this, but for now I'm happy to let you just play with new designs and test them out.

You can test out the Graph+GAN generated levels by running the class HumanSubjectStudy2019Zelda in the edu.southwestern.tasks.gvgai.zelda.study package. Specifically, there is a main method in this file that launches the command:
```MMNEAT.main("zeldaType:generated randomSeed:0 zeldaLevelLoader:edu.southwestern.tasks.gvgai.zelda.level.GANLoader".split(" "));```

If you change the randomSeed you should get a different level. However, if you keep the seed fixed and muck about with the grammar rules, then that should also lead to some new variety.

Give very specific details about any new rules you make ... down the line, we may want to have a list of all the grammar rules similar to what appears at the end of Jake's ZeldaGAN paper ( https://people.southwestern.edu/~schrum2/SCOPE/gutierrez.cec2020.pdf )

**Comment by schrum2** [comment]
The ZeldaHumanSubjectStudy2019GraphGrammar class (your renamed version of Jake's grammar rules) already has a pretty extensive set of rules, but we need many more. The way you should extend the rule set is create a new class (call it something like ExtendedRuleSetGraphGrammar) that extends ZeldaHumanSubjectStudy2019GraphGrammar. All that you need to have in this new class is a constructor that first calls super() and then creates a bunch of new rules.

List LITERALLY every new rule that you add to your new class in this issue thread.

**Comment by schrum2** [comment]
Instead of manually typing out all rules, define some code that takes the grammar rules class and prints out a representation of all of the rules/mappings. Then put that output here.

**Comment by cappsb** [comment]
Made code to print out all the info we need. However, the ending print statement I cannot figure out how to get rid of the strange lettering system after the room type (ie. ENEMY: OMUI)
![image](https://user-images.githubusercontent.com/63438571/82064467-6d380180-9692-11ea-8128-84ed2fda7db6.png)

**Comment by schrum2** [comment]
This issue cannot be considered closed until each possibly pair of non-terminals has at least three different options is can transition to. However, issue #463 is preventing such a rule set from being successfully tested.

**Comment by cappsb** [comment]
When there is an "end" GraphRule, then addNodeToStart and addNodeBetween BOTH simply add a node connected to start. I was under the impression that adding a bunch of things between would just add them in a line, but it just tries to add it next to start, same with add to start.
For the null pointers, it has to do with improper handling of where the end node is, such as when there is a case by itself at the "end" before TREASURE. Since there would not be a proper setNode command, then there is no place to pick up later. I haven't fixed all of them yet because I've run out of time, but many null pointers have been fixed.
Basically, it happens whenever there is a dead-end node that cannot be set as the START for another node set.
